### ISSUE TITLE
Question about Testing REST Plugin

### ISSUE THREAD

**Issue by macanudo527** [eprbell/dali-rp2#42]
I'm about finished with a rough draft of the Binance.com plugin but would like to test it once with real data. How can I run it with my plugin file to see if it pulls the data correctly? I forked the project and pulled it locally, I just can't seem to figure out how to run it. Do I tell python to run the dali_main.py:

```
python3 dali_main.py -s -o output -p test_ binance_config.ini
```
When I run this command, it works on it for a moment then exits. The log file is empty. 

I know I'm missing something obvious here, but I can't seem to find it.

**Comment by eprbell** [comment]
You need to use the development workflow: 
* checkout the code: git clone https://github.com/eprbell/dali-rp2
* setup the environment: https://github.com/eprbell/dali-rp2/blob/main/README.dev.md#setup
* modify the code (e.g. add your new plugin to src/dali/plugin/input/rest)
* run dali normally as described in https://github.com/eprbell/dali-rp2/blob/main/README.md#running

Let me know if you need any more information.

**Comment by macanudo527** [comment]
I tried that initially, but thought I had to do something different to get it to see the config ini file. When I typing either of the following, I get an error:

```
dali -s -o output -p test_ binance_config.ini
dail -s -o output -p test_ test_config.ini
```
This is the error:
```
ERROR: No plugin configuration found in config file. Exiting...
```
I interjected a print command into dali_main.py, so I know I'm running my cloned version. It just doesn't seem to want to read config files. I don't have much experience with Python dev environments like this, so there is a good chance I'm doing something wrong.

I have all the test files, so it should work.

**Comment by eprbell** [comment]
No worries, we'll figure it out. What is the name of the new plugin and where does it reside? Assuming it's binance.py and it sits in `src/dali/plugin/input/rest/`, you need to add a section to your .ini file that looks like this:
```
[dali.plugin.input.rest.coinbase]
account_holder = joe
api_key = <redacted...>
api_secret = <redacted...>
```
The key-value pairs must reflect the signature of the binance plugin constructor, which in this case would have to be:
```
    def __init__(
        self,
        account_holder: str,
        api_key: str,
        api_secret: str,
    ) -> None:
```
Let me know if the above makes sense.

**Comment by eprbell** [comment]
Put it another way, make sure of the following:
* your plugin is named binance.py (or similar)
* your plugin is in `src/dali/plugin/input/rest`
* your plugin class is called `InputPlugin` and derives from `AbstractInputPlugin`: `class InputPlugin(AbstractInputPlugin):`
* your .ini file has a section called `[dali.plugin.input.rest.binance]`, in which the value-pairs map 1-1 the parameters of the plugin constructors (in fact they are used to instantiate the plugin)

Could you paste here a redacted version of your .ini file (make sure it doesn't have personal information, like api_key, api_secret, etc.)?

**Comment by macanudo527** [comment]
I'm pretty sure I did all that. This is the [binance_com.py](https://github.com/macanudo527/dali-rp2/blob/binance_plugin/src/dali/plugin/input/rest/binance_com.py) plugin file in the fork.

Here is the ini:
```
[dali.plugin.input.rest.binance_com]
account_holder = Neal
api_key = <<Redacted>>
api_secret = <<Redacted>>
```

**Comment by macanudo527** [comment]
Ok, after some testing, it seems like I need to add the config directory to the command. The following works:
```
dali -s -o output -p test_ config/binance_config.ini
```

Do I need to add a path somewhere or is this default behavior?

**Comment by eprbell** [comment]
You do need the path to the config file, however you ran into some interesting unexpected behavior: if the .ini file isn't found the `ConfigParser.read()` function fails silently without raising an exception. I just learned that today when reading its documentation to find clues to what you were seeing. So you were getting `ERROR: No plugin configuration found in config file. Exiting...` instead of a `File not found` error. That was unexpected (and confusing) so I fixed it in the latest push. Now if the config file isn't found you get: `Configuration file <config_file> not found`. Thanks for reporting!
